### DISCUSSION TITLE
Better error message for missing methods

### DISCUSSION THREAD

**Discussion by excid3** [josefarias/hotwire_combobox#61] [Ideas]
I tried using Hotwire Combobox this morning and hadn't implemented the `to_combobox_display` method in my model. I got an error and it took me a minute to understand what I needed to do. It would be nice to have a better exception raised here.

This method may need a refactoring to provide different error messages for each usage:

https://github.com/josefarias/hotwire_combobox/blob/d171cb9fa5f3528fd4802a77b1aceabd2876b947/lib/hotwire_combobox/helper.rb#L121-L127

It would be really helpful to new users to provide some thorough error messages in these cases, for example with the above missing method:

```erb
      def hw_call_method_or_proc(object, method_or_proc)
        if method_or_proc.is_a? Proc
          method_or_proc.call object
        else
          object.public_send method_or_proc
        end
      rescue NoMethodError => error
        raise error.exception(<<~MESSAGE
          The #{object.class} class does not respond to `#{method_or_proc}`. This method is used to provide a display value in the Hotwire Combobox. Add this method to your model and return a string for displaying in the options list:
          
              def #{method_or_proc}
                name
              end
        MESSAGE
      end
```

**Comment by josefarias** [comment]
This is a great callout. Thanks!

Tentative fix here: https://github.com/josefarias/hotwire_combobox/pull/63 would love your eyes on it if you have a sec.

**Comment by josefarias** [comment]
PR merged! Happy to refine further, too.
